### DISCUSSION TITLE
Issue with ASP.NET Razor Page

### DISCUSSION THREAD

**Discussion by Bishoy-Saber** [GrapesJS/grapesjs#4883] [Q&A] [answered]
Hi, I need help in some problems I faced, 
1) I faced a problem while using storagemanger ,I created an API and in urlStore passed this API and nothing happens  I wrote autosave:true .
2) I used editor.getProjectData(), and passed the return value through ajax in OnPost function in razor page ASP.NET and every time the value received by the function is null, while when I used getHTML and getCss and do the same the value is send correctly.

so the problem is How to save canvas data (projectData) or Component with assets and styles and pages it in MS SQL Server and how to view them again in canvas so I can edit them again. 

HINT:
1) I have tried so many code and ways to make it work but I found myself posting this question.

Thanks in advance.

**Comment by artf** [comment, marked as answer]
Try to post your init config, did you also check the Network tab in devtools to see what is sent/received?

**Reply by Bishoy-Saber** [reply]
Thanks @artf  for co-operation.
var editor =  grapesjs.init({
                            showOffsets: 1,
                            noticeOnUnload: 0,
                            container: '#gjs',
                            height: '100%',
                            width: '99%',
                            fromElement: true,
                            storageManager:
                            {
                                    type: 'remote',
                                    stepsBeforeSave: 3,
                                    autosave: true,         
                                    autoload: true,
                                    urlStore: 'My-API',
                                    urlLoad:'My-API',
                             },
});

I have check the Network tab nothing is sent and no status code and nothing is received.

**Reply by artf** [reply]
If you're using the latest version of grapesjs your configuration is wrong: https://grapesjs.com/docs/modules/Storage.html#setup-remote-storage
Options for the remote storage are to places inside `config.storageManager.options.remote`

**Reply by Eslamsong** [reply]
@artf  hello, i did the steps that you suggested  and it worked fine the request goes to the API endpoint , but the request project data is not being recognized , i mean that the method's Parameter (data)  is always null

        [HttpPost]
        [Route("{id}")]
        public void AddNewSite(Object data) // data is null , it is not bound with the incoming data
        {

            //here code to be implemented to Save the project data in Database
            
        }

**Reply by Bishoy-Saber** [reply]
Thanks @artf 
It works.
